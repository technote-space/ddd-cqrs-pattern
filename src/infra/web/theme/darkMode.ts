import type { Dispatch, Reducer } from '$/web/shared/store';
import type { IDarkModeContext, IDarkMode, StoreContext } from '$/web/theme/darkMode';
import { useSelector } from 'react-redux';
import { singleton } from 'tsyringe';

@singleton()
export class DarkModeContext implements IDarkModeContext {
  private readonly TOGGLE_DARK_MODE = 'TOGGLE_DARK_MODE';

  public getKey(): string {
    return 'darkMode';
  }

  public getInitialState(): StoreContext {
    return { darkMode: true };
  }

  public getReducerMapObject(): Record<string, Reducer<StoreContext>> {
    return {
      [this.TOGGLE_DARK_MODE]: (store) => ({
        ...store,
        darkMode: !store?.darkMode,
      }),
    };
  }

  public persistTargets(): Array<keyof StoreContext> {
    return ['darkMode'];
  }

  public useDarkMode(): boolean {
    return useSelector((state: { setting: StoreContext }) => state.setting.darkMode);
  }

  public toggleDarkMode(dispatch: Dispatch) {
    dispatch({ type: this.TOGGLE_DARK_MODE });
  }
}

@singleton()
export class DarkMode implements IDarkMode {
  // eslint-disable-next-line
  public useColorModeValue<T>(lightModeValue: T, darkModeValue: T): T {
    return lightModeValue;
  }

  public toggleDarkMode(): () => void {
    return () => {
      //
    };
  }
}
