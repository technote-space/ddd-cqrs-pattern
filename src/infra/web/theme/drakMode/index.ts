import type { useDarkMode as useDarkModeType, useToggleDarkMode as useToggleDarkModeType } from '$/web/theme/darkMode';
import { container } from 'tsyringe';

/* istanbul ignore next */
export const useDarkMode: useDarkModeType = () => container.resolve<useDarkModeType>('useDarkMode')();
/* istanbul ignore next */
export const useToggleDarkMode: useToggleDarkModeType = () => container.resolve<useToggleDarkModeType>('useToggleDarkMode')();
