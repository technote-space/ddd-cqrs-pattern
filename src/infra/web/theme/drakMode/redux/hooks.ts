import type {
  StoreContext,
  useDarkMode as useDarkModeType,
  useToggleDarkMode as useToggleDarkModeType,
} from '$/web/theme/darkMode';
import { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { DarkModeContext } from '.';

export const useDarkMode: useDarkModeType = () => {
  /* istanbul ignore next */
  const selector = (state: { darkMode: StoreContext }) => state.darkMode.darkMode;
  return useSelector(selector);
};

export const useToggleDarkMode: useToggleDarkModeType = () => {
  const dispatch = useDispatch();
  return useCallback(() => {
    dispatch({ type: DarkModeContext.TOGGLE_DARK_MODE });
  }, [dispatch]);
};
