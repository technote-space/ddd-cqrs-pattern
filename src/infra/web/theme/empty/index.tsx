import type { ITheme, Props } from '$/web/theme';
import type { VFC } from 'react';
import { memo } from 'react';
import { singleton } from 'tsyringe';
import { BaseComponent } from '@/web/shared/component';

@singleton()
export class EmptyTheme extends BaseComponent<Props> implements ITheme {
  public constructor() {
    super();
  }

  protected getComponent(): VFC<Props> {
    const component = memo(({ children }) => <div className="empty-theme">
      {children}
    </div>);
    component.displayName = 'EmptyTheme';

    return component;
  }
}
