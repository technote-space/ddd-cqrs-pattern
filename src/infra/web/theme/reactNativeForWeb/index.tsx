import type { ITheme, Props } from '$/web/theme';
import type { VFC } from 'react';
import { singleton } from 'tsyringe';
import { BaseComponent } from '@/web/shared/component';
import View from './view';

@singleton()
export default class ReactNativeForWebTheme extends BaseComponent<Props> implements ITheme {
  public constructor() {
    super();
  }

  protected getComponent(): VFC<Props> {
    return View;
  }
}
