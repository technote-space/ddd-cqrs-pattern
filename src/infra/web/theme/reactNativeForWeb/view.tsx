import type { HooksParams } from './hooks';
import type { FC } from 'react';
import { memo } from 'react';
import { View as RNView } from 'react-native';

const View: FC<HooksParams> = ({ children, styles }) => {
  return <RNView style={styles.root}>
    {children}
  </RNView>;
};

View.displayName = 'ReactNativeForWebThemeView';
export default memo(View);
