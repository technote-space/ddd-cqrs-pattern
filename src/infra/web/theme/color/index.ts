import type { useColor as useColorType, ThemeColor } from '$/web/theme/color';
import { container } from 'tsyringe';
import { useDarkMode } from '@/web/theme/drakMode';

export const useColor: useColorType = (key: keyof ThemeColor) => {
  const themeColor = container.resolve<ThemeColor>('ThemeColor');
  const darkMode = useDarkMode();

  return darkMode ? themeColor[key].dark : themeColor[key].light;
};
