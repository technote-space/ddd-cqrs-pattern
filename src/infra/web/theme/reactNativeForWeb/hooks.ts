import { useMemo } from 'react';
import { StyleSheet } from 'react-native';
import { useColor } from '@/web/theme/color';

export const useHooks = () => {
  const backgroundColor = useColor('background');
  const textColor = useColor('text');
  const styles = useMemo(() => StyleSheet.create({
    root: {
      display: 'flex',
      height: '100%',
      overflow: 'hidden',
      backgroundColor,
      color: textColor,
    },
  }), [backgroundColor, textColor]);

  return {
    styles,
  };
};
export type HooksParams = ReturnType<typeof useHooks>;
