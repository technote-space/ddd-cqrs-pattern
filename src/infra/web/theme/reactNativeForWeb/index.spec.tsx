import renderer from 'react-test-renderer';
import { ReactNativeForWebTheme } from '@/web/theme/reactNativeForWeb';
import * as hooks from './hooks';

describe('ReactNativeForWebTheme', () => {
  it('should return theme', () => {
    jest.spyOn(hooks, 'useHooks').mockReturnValue({ styles: { root: { display: 'flex' } } as never });
    const theme = new ReactNativeForWebTheme();
    const tree = renderer.create(theme.render({ children: <div>test</div> }));
    expect(tree).toMatchSnapshot();
  });
});
