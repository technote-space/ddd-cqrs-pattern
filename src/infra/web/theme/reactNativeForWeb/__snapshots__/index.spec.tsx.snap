// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactNativeForWebTheme should return theme 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div>
    test
  </div>
</div>
`;
